From b55c97d7930caec06d559faae4684761f258fd0f Mon Sep 17 00:00:00 2001
From: Michael Soegtrop <7895506+MSoegtropIMC@users.noreply.github.com>
Date: Thu, 30 Apr 2020 13:50:08 +0200
Subject: [PATCH 10/12] Added open source build to makefile

---
 Makefile | 36 +++++++++++++++++++++++++++++++++---
 1 file changed, 33 insertions(+), 3 deletions(-)

diff --git a/Makefile b/Makefile
index df5fb03f..48770ebd 100644
--- a/Makefile
+++ b/Makefile
@@ -57,10 +57,14 @@ FLOCQ=\
   Relative.v Sterbenz.v Round_odd.v Double_rounding.v \
   Binary.v Bits.v
 
+# Architecture files (in respective architecture folder)
+
+ARCHFILES=Archi.v
+
 # General-purpose libraries (in lib/)
 
 VLIB=Axioms.v Coqlib.v Intv.v Maps.v Heaps.v Lattice.v Ordered.v \
-  Iteration.v Zbits.v Integers.v Archi.v IEEE754_extra.v Floats.v \
+  Iteration.v Zbits.v Integers.v IEEE754_extra.v Floats.v \
   Parmov.v UnionFind.v Wfsimpl.v \
   Postorder.v FSetAVLplus.v IntvSets.v Decidableplus.v BoolEqual.v
 
@@ -101,6 +105,8 @@ BACKEND=\
   Bounds.v Stacklayout.v Stacking.v Stackingproof.v \
   Asm.v Asmgen.v Asmgenproof0.v Asmgenproof1.v Asmgenproof.v
 
+BACKEND_OPEN_SOURCE=Cminor.v
+  
 # C front-end modules (in cfrontend/)
 
 CFRONTEND=Ctypes.v Cop.v Csyntax.v Csem.v Ctyping.v Cstrategy.v Cexec.v \
@@ -110,6 +116,8 @@ CFRONTEND=Ctypes.v Cop.v Csyntax.v Csem.v Ctyping.v Cstrategy.v Cexec.v \
   Cshmgen.v Cshmgenproof.v \
   Csharpminor.v Cminorgen.v Cminorgenproof.v
 
+CFRONTEND_OPEN_SOURCE=Clight.v ClightBigstep.v Cop.v Csem.v Cstrategy.v Csyntax.v Ctypes.v Ctyping.v
+
 # Parser
 
 PARSER=Cabs.v Parser.v
@@ -126,9 +134,17 @@ DRIVER=Compopts.v Compiler.v Complements.v
 
 # All source files
 
-FILES=$(VLIB) $(COMMON) $(BACKEND) $(CFRONTEND) $(DRIVER) $(FLOCQ) \
+FILES=$(ARCHFILES) $(VLIB) $(COMMON) $(BACKEND) $(CFRONTEND) $(DRIVER) $(FLOCQ) \
   $(MENHIRLIB) $(PARSER)
 
+# All open source source files (in the order given in LICENSE)
+
+# ATTENTION: this also includes ./x86/Builtins1.vo - which is not open source but many files depend on it
+
+FILES_OPEN_SOURCE=$(VLIB) $(COMMON) $(CFRONTEND_OPEN_SOURCE) $(BACKEND_OPEN_SOURCE) $(PARSER) Clightdefs.v $(ARCHFILES) 
+
+# These files have non open dependencies: extractionMachdep.v, extraction.v
+
 # Generated source files
 
 GENERATED=\
@@ -153,6 +169,8 @@ endif
 
 proof: $(FILES:.v=.vo)
 
+proof_open_source: $(FILES_OPEN_SOURCE:.v=.vo) compcert.config
+
 # Turn off some warnings for compiling Flocq
 flocq/%.vo: COQCOPTS+=-w -compatibility-notation
 
@@ -181,7 +199,7 @@ runtime:
 
 FORCE:
 
-.PHONY: proof extraction runtime FORCE
+.PHONY: proof proof_open_source extraction runtime FORCE
 
 documentation: $(FILES)
 	mkdir -p doc/html
@@ -278,6 +296,18 @@ ifeq ($(INSTALL_COQDEV),true)
 	@(echo "To use, pass the following to coq_makefile or add the following to _CoqProject:"; echo "-R $(COQDEVDIR) compcert") > $(DESTDIR)$(COQDEVDIR)/README
 endif
 
+# ToDo: copy exactly the files in FILES_OPEN_SOURCE as soon as the issue with Builtins1 ins fixed
+install_open_source:
+ifeq ($(INSTALL_COQDEV),true)
+	install -d $(DESTDIR)$(COQDEVDIR)
+	for d in $(DIRS); do \
+          install -d $(DESTDIR)$(COQDEVDIR)/$$d && \
+          install -m 0644 $$d/*.vo $(DESTDIR)$(COQDEVDIR)/$$d/; \
+	done
+	install -m 0644 ./VERSION $(DESTDIR)$(COQDEVDIR)
+	install -m 0644 ./compcert.config $(DESTDIR)$(COQDEVDIR)
+	@(echo "To use, pass the following to coq_makefile or add the following to _CoqProject:"; echo "-R $(COQDEVDIR) compcert") > $(DESTDIR)$(COQDEVDIR)/README
+endif
 
 clean:
 	rm -f $(patsubst %, %/*.vo*, $(DIRS))
-- 
2.24.2 (Apple Git-127)

