From b46c0c01379da17dd96fc0cb8f0458100b7b1e5e Mon Sep 17 00:00:00 2001
From: Xavier Leroy <xavier.leroy@college-de-france.fr>
Date: Mon, 4 May 2020 10:51:47 +0200
Subject: [PATCH 08/12] Update the list of dual-licensed files

Closes: #351
---
 LICENSE | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/LICENSE b/LICENSE
index 5a7ae79f..61b84219 100644
--- a/LICENSE
+++ b/LICENSE
@@ -46,8 +46,8 @@ option) any later version:
 
   all files in the exportclight/ directory
 
-  the Archi.v, CBuiltins.ml, and extractionMachdep.v files
-  in directories arm, powerpc, riscV, x86, x86_32, x86_64
+  the Archi.v, Builtins1.v, CBuiltins.ml, and extractionMachdep.v files
+  in directories aarch64, arm, powerpc, riscV, x86, x86_32, x86_64
 
   extraction/extraction.v
 
-- 
2.24.2 (Apple Git-127)

