From 40fdac14428ef787ec94c137baea11fcfec6cb36 Mon Sep 17 00:00:00 2001
From: Michael Soegtrop <7895506+MSoegtropIMC@users.noreply.github.com>
Date: Tue, 5 May 2020 17:10:06 +0200
Subject: [PATCH 12/12] Use platform supplied menhirlib as suggested by
 jhjourdan

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 48770ebd..c4130630 100644
--- a/Makefile
+++ b/Makefile
@@ -265,7 +265,7 @@ driver/Version.ml: VERSION
 
 cparser/Parser.v: cparser/Parser.vy
 	@rm -f $@
-	$(MENHIR) --coq --coq-lib-path compcert.MenhirLib --coq-no-version-check cparser/Parser.vy
+	$(MENHIR) --coq cparser/Parser.vy
 	@chmod a-w $@
 
 depend: $(GENERATED) depend1
-- 
2.24.2 (Apple Git-127)

